_python/site_parser.py

```diff
@@ -0,0 +1,101 @@
+import psycopg2 as ps
+import bs4
+import pandas as pd
+import requests
+import os
+import logging
+from dotenv import load_dotenv
+
+logging.basicConfig(level=logging.INFO, format='%(asctime)s - %(message)s')
+load_dotenv()
+
+class Webscrapper:
+
+    def __init__(self, url: str) -> None:
+        self.url = url
+
+    @staticmethod
+    def _start_connections() -> ps.extensions.connection:
+        try:
+            connection = ps.connect(
+                host = os.getenv('DB_HOST'),
+                port = os.getenv('DB_PORT'),
+                dbname = os.getenv('DB_NAME'),
+                password = os.getenv('PASSWORD'),
+                user = os.getenv('DB_USER')
+            )
+
+            return connection
+        except Exception as e:
+            logging.info(f"Error: {e}")
+
+    @staticmethod
+    def _get_data(url: str) -> list[tuple[str, str, str]]:
+        content = requests.request("get", url)
+        soup = bs4.BeautifulSoup(content.text, "html.parser")
+
+        all_lists = soup.find_all('ul', id="", recursive=True)
+        res = []
+
+        for ele in all_lists:
+            if not ele.get('class') and len(ele) == 21:
+               text_list = ele.text.replace("\xa0", "").split("\n")
+
+        for line in text_list:
+            splitted = line.split("— ")
+            splitter = splitted[1].index(",")
+
+            res.append((splitted[0], splitted[1][:splitter], splitted[1][splitter+2:]))
+
+        return res
+    
+    def _construct_df(self) -> pd.DataFrame:
+        site_data = self._get_data(self.url)
+
+        dict_data = {
+            "Date": [f"{date} 2000" for date, _, _ in site_data],
+            "Name": [name for _, name, _ in site_data],
+            "Info": [info for _, _, info in site_data],
+        }
+
+        df = pd.DataFrame.from_dict(dict_data)
+
+        return df
+    
+    def website_data_to_db(self) -> None:
```

Comment:
Can we instead use [this](https://pandas.pydata.org/docs/reference/api/pandas.DataFrame.to_sql.html)?